client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -51,6 +51,37 @@ interface IAnalysisWizard {
   isOpen: boolean;
 }
 
+const determineMode = (
+  applications: Application[]
+):
+  | "binary"
+  | "source-code-deps"
+  | "source-code"
+  | "binary-upload"
+  | undefined => {
+  // If only one application is selected
+  if (applications.length === 1) {
+    const { repository, binary } = applications[0];
+    // If the application has a repository or both repository and binary definitions, return "source-code-deps"
+    // If the application has only binary definitions, return "binary"
+    // If no definitions are present, return undefined
+    return repository || (repository && binary)
+      ? "source-code-deps"
+      : binary && binary !== ""
+      ? "binary"
+      : undefined;
+  }
+  // Check if all selected applications have only binary definitions
+  const allBinary = applications.every((app) => app.binary);
```

Comment:
1. `applications.every((app) => app.binary)` will work also for one element array -  it seems that you could delete the single-app case above and rely only on multiple-app case.
2. the code is self-documenting - please leave the comments only in places where variable names are not enough

Reply:
I fixed it